File: web/types/audience.ts
import { Idea } from "./idea";

export interface AudienceStats {
  totalMembers: number;
  conversionRate: number;
  totalSubscribers: number;
  newSubscribers: number;
  newSubscribersChange: number;
  averageConversionRate: number;
  conversionRateChange: number;
  activeIdeas: number;
}

export interface AudienceMember {
  userId: string;
  email: string;
  name?: string;
  signupTime: string;
  ideaId: string; // Which idea they signed up for
  mvpId: string; // The MVP they are associated with
  idea?: Partial<Idea>;
  engaged: boolean;
  converted: boolean; // Whether they signed up for the idea
  // Optional tracking data
  lastActive?: string;
  visits?: number;
}

File: api/internal/dto/response/audience.go
package response

import "github.com/google/uuid"

type AudienceResponse struct {
	Audiences []*Audience       `json:"audiences"`
	Stats     *AudienceStats    `json:"stats"`
	Total     int64             `json:"total"`
	Metrics   []AudienceMetrics `json:"metrics"`
}

type Audience struct {
	UserID     string    `json:"userId"`
	IdeaID     uuid.UUID `json:"ideaId"`
	IdeaTitle  string    `json:"ideaTitle"`
	SignupTime string    `json:"signupTime"`
}

type AudienceStats struct {
	TotalSubscribers      int64   `json:"totalSubscribers"`
	NewSubscribers        int64   `json:"newSubscribers"`
	NewSubscribersChange  float64 `json:"newSubscribersChange"`
	AverageConversionRate float64 `json:"averageConversionRate"`
	ConversionRateChange  float64 `json:"conversionRateChange"`
	ActiveIdeas           int64   `json:"activeIdeas"`
}

type AudienceMetrics struct {
	IdeaTitle string `json:"name"`
	Count     int64  `json:"value"`
}
